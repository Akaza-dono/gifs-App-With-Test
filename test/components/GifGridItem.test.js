import { render, screen } from "@testing-library/react"
import { GifGridItem } from "../../src/components/GifGridItem"

describe('pruebas en GifGridItem',()=>{

    const title = 'goku'
    const url = 'https://media1.giphy.com/media/hs67xo8fGYfx5KlBgV/giphy.gif?cid=5ab5ee72dnc4lw90y5gk9x79jsfz0i2x9d64k3tjydq1y27q&ep=v1_gifs_search&rid=giphy.gif&ct=g'

    test('GifGridItem debe de hacer match con el snapshots',()=>{
        const {container} = render(<GifGridItem title={title} url={url}/>)
        expect(container).toMatchSnapshot();
    })

    test('debe de mostrar la imagen con su alt',()=>{
        render(<GifGridItem title={title} url={url}/>);
        expect(screen.getByRole('img').src).toBe(url) 
    });

    test('debe de mostrar el titulo en el componente',()=>{
        render(<GifGridItem title={title} url={url}/>);
        expect(screen.getByText(title)).toBeTruthy()
    })

})