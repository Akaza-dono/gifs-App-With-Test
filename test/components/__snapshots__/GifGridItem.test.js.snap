// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en GifGridItem GifGridItem debe de hacer match con el snapshots 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="goku"
      src="https://media1.giphy.com/media/hs67xo8fGYfx5KlBgV/giphy.gif?cid=5ab5ee72dnc4lw90y5gk9x79jsfz0i2x9d64k3tjydq1y27q&ep=v1_gifs_search&rid=giphy.gif&ct=g"
    />
    <p>
      goku
    </p>
  </div>
</div>
`;
